test/test-commands.ts

```diff
@@ -85,6 +85,14 @@ describe("splitListItem", () => {
            doc(ul(li(p("one")), li(p("<a>")), li(p("two"))))))
 })
 
+describe("splitListItemKeepMarks", () => {
+   let splitWithMarks = splitListItemKeepMarks(schema.nodes.list_item)
+
+   it("preserves marks after splitting", () => 
+     apply(doc(ul(li(p(strong("foo<a>bar"))))), splitWithMarks, 
+           doc(ul(li(p(strong("foo"))), li(p(strong("bar")))))))
+})
```

Comment:
This looks like it doesn't actually test the preserving---it doesn't look at what the stored marks are after the command runs.